CMakeLists.txt

```diff
@@ -521,67 +521,71 @@ if (NOT EMSCRIPTEN)
+      function(c_api_example NAME)
+        set(EXENAME wasm-c-api-${NAME})
+        add_executable(${EXENAME} third_party/wasm-c-api/example/${NAME}.c)
+        if (NOT COMPILER_IS_MSVC)
+          set_target_properties(${EXENAME} PROPERTIES COMPILE_FLAGS "-std=gnu11 -Wno-pointer-to-int-cast")
+        endif ()
+        target_link_libraries(${EXENAME} wasm Threads::Threads)
+        add_custom_target(${EXENAME}-copy-to-bin ALL
+          COMMAND ${CMAKE_COMMAND} -E make_directory ${WABT_SOURCE_DIR}/bin
+          COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${EXENAME}> ${WABT_SOURCE_DIR}/bin/
+          COMMAND ${CMAKE_COMMAND} -E copy ${WABT_SOURCE_DIR}/third_party/wasm-c-api/example/${NAME}.wasm $<TARGET_FILE_DIR:${EXENAME}>/
+          COMMAND ${CMAKE_COMMAND} -E copy ${WABT_SOURCE_DIR}/third_party/wasm-c-api/example/${NAME}.wasm ${WABT_SOURCE_DIR}/bin/
+          DEPENDS ${EXENAME}
+        )
+        add_dependencies(run-c-api-tests ${EXENAME})
+      endfunction()
+
+      c_api_example(callback)
+      c_api_example(finalize)
+      c_api_example(global)
+      c_api_example(hello)
+      c_api_example(hostref)
+      c_api_example(multi)
+      c_api_example(memory)
+      c_api_example(reflect)
+      c_api_example(serialize)
+      c_api_example(start)
+      c_api_example(table)
+      c_api_example(trap)
+      if (NOT WIN32)
+        # depends on pthreads
+        set(THREADS_PREFER_PTHREAD_FLAG ON)
+        find_package(Threads REQUIRED)
+        c_api_example(threads)
       endif ()
-      target_link_libraries(${EXENAME} wasm Threads::Threads)
-      add_custom_target(${EXENAME}-copy-to-bin ALL
-        COMMAND ${CMAKE_COMMAND} -E make_directory ${WABT_SOURCE_DIR}/bin
-        COMMAND ${CMAKE_COMMAND} -E copy $<TARGET_FILE:${EXENAME}> ${WABT_SOURCE_DIR}/bin/
-        COMMAND ${CMAKE_COMMAND} -E copy ${WABT_SOURCE_DIR}/third_party/wasm-c-api/example/${NAME}.wasm $<TARGET_FILE_DIR:${EXENAME}>/
-        COMMAND ${CMAKE_COMMAND} -E copy ${WABT_SOURCE_DIR}/third_party/wasm-c-api/example/${NAME}.wasm ${WABT_SOURCE_DIR}/bin/
-        DEPENDS ${EXENAME}
-      )
-      add_dependencies(run-c-api-tests ${EXENAME})
-    endfunction()
-
-    c_api_example(callback)
-    c_api_example(finalize)
-    c_api_example(global)
-    c_api_example(hello)
-    c_api_example(hostref)
-    c_api_example(multi)
-    c_api_example(memory)
-    c_api_example(reflect)
-    c_api_example(serialize)
-    c_api_example(start)
-    c_api_example(table)
-    c_api_example(trap)
-    if (NOT WIN32)
-      # depends on pthreads
-      set(THREADS_PREFER_PTHREAD_FLAG ON)
-      find_package(Threads REQUIRED)
-      c_api_example(threads)
-    endif ()
+    else()
+      message(FATAL_ERROR "Python 3 is required for wabt testing. Please install it. \nIf you want to proceed without installing python 3, either add -no-tests to make target (example: make clang-debug-no-tests) or run cmake with -DBUILD_TESTS=OFF.")
+    endif(PYTHONINTERP_FOUND)
```

Comment:
Seem like you could just put this error in an if an "if (NOT PYTHONINTERP_FOUND) .. endif()" above that the rest of the file would be left unchanged?

Reply:
Done